lib/class-wp-rest-menu-items-controller.php

```diff
@@ -929,6 +999,41 @@ protected function prepare_items_query( $prepared_args = array(), $request = nul
 		return $query_args;
 	}
 
+	/**
+	 * Checks whether current user can assign all terms sent with the current request.
+	 *
+	 * @param WP_REST_Request $request The request object with post and terms data.
+	 *
+	 * @return bool Whether the current user can assign the provided terms.
+	 */
+	protected function check_assign_terms_permission( $request ) {
+		$taxonomies = wp_list_filter( get_object_taxonomies( $this->post_type, 'objects' ), array( 'show_in_rest' => true ) );
+		foreach ( $taxonomies as $taxonomy ) {
+			$base = ! empty( $taxonomy->rest_base ) ? $taxonomy->rest_base : $taxonomy->name;
+
+			if ( ! isset( $request[ $base ] ) ) {
+				continue;
+			}
+
+			foreach ( (array) $request[ $base ] as $term_id ) {
+				if ( ! $term_id ) {
+					continue;
+				}
+
+				// Invalid terms will be rejected later.
+				if ( ! get_term( $term_id, $taxonomy->name ) ) {
+					continue;
+				};
+
+				if ( ! current_user_can( 'assign_term', (int) $term_id ) ) {
+					return false;
+				}
+			}
+		}//end foreach
```

Comment:
Ditto as above.

Reply:
Fixed in https://github.com/WP-API/menus-endpoints/pull/41/commits/006a52b4d42711a0778f661225c090ad24b22db9